src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,372 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.I2c;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Binding that exposes an INA219 Bidirectional Current/Power monitor.
+    /// 
+    /// The INA219 is a high-side current shunt and power monitor with an I2C interface.
+    /// The INA219 monitors both shunt drop and supply voltage, with programmable conversion
+    /// times and filtering. A programmable calibration value, combined with an internal multiplier,
+    /// enables direct readouts in amperes. An additional multiplying register calculates power in watts.
+    /// <see href="http://www.ti.com/lit/ds/symlink/ina219.pdf"/>
+    /// </summary>
+    public class Ina219 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _disposeI2cDevice = false;
+        private ushort _calibrationValue;
+        private Ina219AdcResolutionOrSamples _busAdcResSamp;
+        private Ina219AdcResolutionOrSamples _shuntAdcResSamp;
+        private float _currentLsb;
+
+        // These are possitive ranges in volts associated with each gain setting of the Programable Gain Controller.
+        private readonly Dictionary<Ina219PgaSensitivity, float> _sensitivityVoltage = new Dictionary<Ina219PgaSensitivity, float>()
```

Comment:
if this is not being modified this can be safely static (if it's modified then it won't be thread safe)

Reply:
Actually not being used so removed.